earth_enterprise/src/scons/khEnvironment.py

```diff
@@ -159,6 +162,192 @@ def EmitBuildDateStrfunc(target, build_date):
   return 'EmitBuildDate(%s, %s)' % (target, build_date)
 
 
+def EmitVersionHeaderFunc(target, backupFile):
+  """Emit version information to the target file."""
+
+  versionStr = GetVersion(backupFile)
+  longVersionStr = GetLongVersion(backupFile)
+
+  fp = open(target, 'w')
+  fp.writelines(['// DO NOT MODIFY - auto-generated file\n',
+                 'extern const char *const GEE_VERSION = "' +
+                 versionStr + '";\n',
+                 'extern const char *const GEE_LONG_VERSION = "' +
+                 longVersionStr + '";\n'
+                ])
+  fp.close()
+
+
+def EmitVersionHeaderStrfunc(target, backupFile):
+  return 'EmitVersionHeader(%s, %s)' % (target, backupFile)
+  
+  
+def EmitVersionFunc(target, backupFile):
+  """Emit version information to the target file."""
+
+  versionStr = GetVersion(backupFile)
+
+  fp = open(target, 'w')
+  fp.write(versionStr)
+  fp.close()
+  
+  fp = open(backupFile, 'w')
+  fp.write(versionStr)
+  fp.close()
+
+
+def EmitVersionStrfunc(target, backupFile):
+  return 'EmitVersion(%s, %s)' % (target, backupFile)
+  
+  
+def EmitLongVersionFunc(target, backupFile):
+  """Emit version information to the target file."""
+
+  versionStr = GetLongVersion(backupFile)
+
+  fp = open(target, 'w')
+  fp.write(versionStr)
+  fp.close()
+
+
+def EmitLongVersionStrfunc(target, backupFile):
+  return 'EmitLongVersion(%s, %s)' % (target, backupFile)
+  
+
+def GetLongVersion(backupFile):
+    """Create a detailed version string based on the state of
+       the software, as it exists in the repository."""
+    
+    if CheckGitAvailable():
+        return GitGeneratedLongVersion()
+
+    # Without git, must use the backup file to create a string.
+    base = ReadBackupVersionFile(backupFile)
+    date = datetime.utcnow().strftime("%Y%m%d%H%M")
+    
+    return '-'.join([base, date])
+
+
+def GetVersion(backupFile):
+    """As getLongVersion(), but only return the leading *.*.* value."""
+    try:
+        raw = GetLongVersion(backupFile)
+        final = raw.split("-")[0]
+    except Exception:
+        return "Version Error"
+
+    return final
+  
+
+def CheckGitAvailable():
+    """Try the most basic of git commands, to see if there is
+       currently any access to a repository."""
+    
+    try:
+        subprocess.check_output(['git', 'status'])
```

Comment:
Consider using GitPython, or write a comment about why it's not suitable for the environments we support.